C++/include/algorithm/dynamic_programming/longest_subsequence.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef LONGEST_SUBSEQUENCE
+#define LONGEST_SUBSEQUENCE
+
+/*
+Made by Cole Severson
+<Longest Subsequence>
+---------------------
+This algorithm finds either the longest increasing (or decreasing depending on the order varialbe) subsequence in a sequence of numbers
+It follows the principals of dynamic programming
+
+Time complexity
+----------------
+O(n logn) where n is the number of elements in the initial list
+
+Space complexity
+---------------
+O(n)
+
+*/
+
+
+#include <iostream>
+#include <vector>
+
+
+
+
+int longest_subsequence(std::vector<int>& input, int order = 1, const bool to_show_state = true) {//order of 1 means increasing subsequence, -1 for decreasing subsequence
+    int elements = input.size();//number of elements in the vector
+    
+    if(elements == 1 || elements == 0) 
```

Comment:
Add space between if and parens. And include a braces after the if statement, like so:

```
if (condition) {
    // code
}
```